addons/account/models/account_move.py

```diff
@@ -1653,6 +1653,10 @@ def _inverse_partner_id(self):
 
     @api.onchange('company_id')
     def _inverse_company_id(self):
+        # When the onchange will be called for changing the company_id at this time
+        # if there would be no record for res.company().it throws the Singleton error.
+        if not self.company_id:
+            raise ValidationError(_("Select the Company"))
```

Comment:
- Regarding the code: 
  The record set `self` might contain several records. 
  We need to ensure everysingle one of them has the `company_id` set
- Regarding the comment:
  That's what we said on discord:
  > you need to add a comment explaining why we are doing it there **instead of the constrains**.
- The validation error isn't precise enough. "Select the company" is quite vague.

```suggestion
        for move in self:
            # This can't be caught by a python constraint as it is only triggered at save and the compute method that
            # needs this data to be set correctly are triggered before saving
            if not self.company_id:
                raise ValidationError(_("We can't leave this document without any company. Please select a company for this document."))

```